### PR TITLE
Sphinx various fixing of failing commands

### PR DESCRIPTION
**Pull request by SkySkimmer**
Doc-only #9501 (ie without the changes to coqtop which support checking that new failures don't get introduced) for easy backport.

### CODE REVIEW

File: doc/sphinx/addendum/type-classes.rst [hunk 1]
```diff
@@ -564,12 +559,12 @@ Settings
    This flag allows to switch the behavior of instance declarations made through
    the Instance command.
 
-   + When it is on (the default), instances that have unsolved holes in
+   + When it is off (the default), they fail with an error instead.
```
Comment (Zimmi48) [inline doc/sphinx/addendum/type-classes.rst]: The commit fixing this cannot be backported (since it is not the same in master and v8.9) so this should be moved to another PR. Sorry about that :disappointed:

**Review by Zimmi48 — COMMENTED** [review]
Thanks! I just have one small comment before I can merge this.

File: doc/sphinx/addendum/type-classes.rst (hunk 1 above)
Comment (SkySkimmer) [reply doc/sphinx/addendum/type-classes.rst]: Removed
